src/lib/operators/join_hash.cpp

```diff
@@ -620,6 +112,11 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
     const auto build_relation_size = _left->get_output()->row_count();
     const auto probe_relation_size = _right->get_output()->row_count();
 
+    if (radix_bits.has_value()) {
+      _radix_bits = radix_bits.value();
```

Comment:
Please extract the calculation of the radix bits into its own function. That way we can save us the return (which people might miss) and simply write 

```
if (radix_bits.has_value()) {
  _radix_bits = radix_bits.value();
} else {
  _radix_bits = _calculate_radix_bits();
}
```

Reply:
Good point. Much better to read.